src/resmom/linux/alps.c

```diff
@@ -3187,12 +3189,28 @@ inventory_to_vnodes(basil_response_t *brp)
 			default:
 				continue;
 		}
-		for (seg=node->segments; seg; seg=seg->next) {
-			char	vname[128];
 
-			snprintf(vname, sizeof(vname), "%s_%ld_%d",
-				mpphost, node->node_id, seg->ordinal);
-			vname[sizeof(vname)-1]='\0';
+		totcpus = 0;
+		totmem = 0;
+
+		/* Only do this for nodes that have accelerators. */
+                if (node->accelerators) {
+                        for (accel=node->accelerators, totaccel=0;
+                                accel; accel=accel->next) {
+                                if (accel->state == basil_accel_state_up)
+                                        /* Only count them if the state is UP */
+                                        totaccel++;
+                        }
+                }
```

Comment:
By moving it outside of the segment loop we can save ourselves a few cycles this way.  In the BASIL 1.2 XML that we parse, the accelerator information is provided per node, not per segment.  So when the code was further down we were unnecessarily calculating the same accelerator information again and again per segment.